### DISCUSSION TITLE
FNIRSI GC-01: Download firmware from device

### DISCUSSION THREAD

**Discussion by martin-machacek** [Gissio/radpro#24] [Q&A] [answered]
Hi, 
is it possible to download the original firmware from the microcontroller of FNIRSI GC-01? As you know the device doesn't show the fimware version and I would like to back up the exact version my device had before trying radpro (at least until stable release is available).
It may be a dumb question, I only play with arduino etc...

P.S.: Thank you for such a great project. I can send you photo of my PCB if it helps.
Thank you

**Comment by Gissio** [comment, marked as answer]
Unfortunately the bootloader protects the flash memory, so there is no way to do so.

**Reply by martin-machacek** [reply]
Never mind, I will just download the beta and join the testers team then. Thanks

**Comment by jackhen** [comment]
Ok Guys, here is the foolproff to get the FW loaded. Works everytime.....
After you connect your GC-01 via usb to your PC you "hold down" the power switch. As long as you hold it down, you will see the disk appear on your PC. So hold it down, then drag the repo onto your disk icon. Release the power button when you see the GC-01 booting. Takes only a few seconds. And it doesn't matter the USB protocol. It just works.

**Reply by Gissio** [reply]
It is very understandable that you are happy with your findings! But please don't flood the discussions.
